contrib/simplan/simplan.nit

```diff
@@ -0,0 +1,629 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A simple planning problem solver for a robot that delivers parcels
+module simplan
+
+import ai::search
+
+private import simplan_lexer
+private import simplan_parser
+
+# The description of the planing problem to solve
+class PlanProblem
+	super SearchProblem[State, Action]
+
+	# Some parameters of the problem
+
+	# The speed on the road.
+	# Used for the action `Drive`
+	var speed = 1.0
+
+	# Cost of loading a parcel in a robot
+	# Used for the action `Load`
+	var duration_loading = 30.0
+
+	# Cost of loading a parcel in a robot
+	# Used for the action `Unload`
+	var duration_unloading = 30.0
+
+	# How many parcels can a robot transport at the same time?
+	var capacity = 2
+
+	# The name of the robot
+	var robot_name = "r0"
+
+	# Locations on the map, by their names
+	var locations = new HashMap[String, Location]
+
+	# All the parcels, by their `id`
+	var parcels = new Array[Parcel]
+
+	# All the parcels, by their names
+	var parcel_by_name = new HashMap[String, Parcel]
+
+	# A big number, usually +inf
```

Comment:
Is this used when computing the shortest path? Documentation can be improved to explain why this is usefuull.